specs/deneb/beacon-chain.md

```diff
@@ -293,9 +293,13 @@ def verify_and_notify_new_payload(self: ExecutionEngine,
     """
     Return ``True`` if and only if ``new_payload_request`` is valid with respect to ``self.execution_state``.
     """
+    # [Modified in Deneb:EIP4788]
     execution_payload = new_payload_request.execution_payload
     parent_beacon_block_root = new_payload_request.parent_beacon_block_root
 
+    if b'' in execution_payload.transactions:
+        return False
```

Comment:
I think I understand the implications of the PR and agree that zero-length transactions should be rejected, but I'm not certain that `verify_and_notify_new_payload` is best location for such a check. I feel like `notify_new_payload` via the Engine API should return false if it sees this.

Reply:
I think `notify_new_payload` is not an Engine API thing anymore. In fact, `verify_and_notify_new_payload` is an Engine API. AFAIU, the reason that we have the logic of `verify_and_notify_new_payload` in the consensus-specs is because we want to show what the function does, but it doesn't mean that it's an CL thing.

So I think putting it in `verify_and_notify_new_payload` already makes sense because it's a "verify" thing.

Reply:
Hmm okay, well in that case I guess it doesn't hurt to add. Let's do it!

@etan-status could you merge in dev & make similar changes to electra?

Reply:
Yep, this is a mock implementation for the minimal EL checks that have to be done.

- https://github.com/ethereum/execution-apis/pull/573

This is the matching spec that describes the additional check for zero length transactions.